### DISCUSSION TITLE
How to replace text of a single file and keep all files?

### DISCUSSION THREAD

**Discussion by GCheung55** [newren/git-filter-repo#292] [Q&A] [answered]
How do I go about replacing text of a single file and keep all the files?

I've tried the following command, however only `lib/foo/bar.js` is edited and kept while all files are removed:
```
git-filter-repo ./ --path 'lib/foo/bar.js' --replace-text '../expression.txt'
```

`bfg-ish` is able to target a file that matches the file basename and keep all history, but would act on all files matching the basename.

**Comment by newren** [comment, marked as answer]
See the contrib/filter-repo-demos/lint-history script in the repository, particularly the `--relevant` option.  That'll let you do modifications to a single file or class of files while leaving all others present.

Besides the fact that filter-repo treats `--path` as select-these-paths-to-keep rather than select-these-paths-to-operate-on, your filter-repo command also has a stray `./` argument on it.  What was your intended purpose with that argument?

**Reply by newren** [reply]
You could also use the new --file-info-callback instead of lint-history.
